src/main/java/com/selfxdsd/todocli/TodoVisitor.java

```diff
@@ -89,8 +105,19 @@ public FileVisitResult postVisitDirectory(
             final Path dir, final IOException exc
     ) throws IOException {
         if (dir.equals(root)) {
-            // scanning root has finished.
-            this.serializer.serialize();
+            try {
+                // scanning root has finished.
+                this.service.shutdown();
+                this.service.awaitTermination(5, TimeUnit.MINUTES);
+                if (this.hasErrors.get()) {
+                    throw new IOException("Could not serialize todos; something"
+                        + " went wrong while parsing a file for todos.");
+                } else {
+                    this.serializer.serialize();
```

Comment:
@amihaiemil now serialization is done only if there are no errors while parsing the src files

Reply:
@criske no, please remove this and we'll do it in separate ticket if necessary. I want the parsing to take place even if there is an error in one of the todos. 

That's one of the things I don't like about the original [pdd](https://github.com/yegor256/pdd) -- it crashes and reports no tickets if it finds an error, even if only 1 todo is wrong.